### ISSUE TITLE
MlpModule overfits on validation data

### ISSUE THREAD

**Issue by truebluejason** [keras-team/autokeras#649]
### Bug Description
Currently, model_trainer accepts train_loader and test_loader. However, we are passing the same train_loader and test_loader we specify in mlp_module.fit() over and over again during the search phase. Since AutoKeras evaluates more than 100+ models on the same validation set and produces metrics based on this, I noticed that sometimes my test error actually decreases the more I train AutoKeras.

### Reproducing Steps
* Run mlpmodule.fit / mlpmodule.predict on a same dataset and compute the Accuracy for 1, 2, 3, and 6 hours.

### Expected Behavior
The test error increases slightly.

### Setup Details
Include the details about the versions of:
 - OS type and version: Ubuntu 18.04.1
 - Python: 3.6
 - autokeras: 0.4.0

### Additional context
I suggest giving mlpmodule.fit a single dataloader and use cross validation for computing the loss metric.
